manifests/agent/config.pp

```diff
@@ -36,12 +36,10 @@
     }
   }
 
-  # we need to store this in a variable, because older puppet doesn't
-  # like resource{function(): ... }
-  $additional_settings_keys = keys($::puppet::agent_additional_settings)
-  puppet::config::additional_settings{ $additional_settings_keys:
-    hash     => $::puppet::agent_additional_settings,
-    resource => '::puppet::config::agent',
+  $::puppet::agent_additional_settings.each |$key,$value| {
+    puppet::config::agent {
+      $key: value => $value,
```

Comment:
Nit: typically we place the name of the resource at the same line as the resource. So ```$key``` should be on the same line as ```puppet::config::agent```.